Tests/Tests/DDFileLoggerTests.m

```diff
@@ -39,6 +39,12 @@ - (void)tearDown {
     [super tearDown];
     
     [DDLog removeAllLoggers];
+    // We need to sync all involved queues to wait for the post-removal processing of the logger to finish before deleting the files.
+    dispatch_sync([DDLog loggingQueue], ^{
```

Comment:
Maybe we should add check for distinct queues?
Because double syncing isn't safe on one queue.
Maybe we should add check for distinct queues?
Because double syncing isn't safe on one queue.

Something like
```objective-c
if (queue != self->logger.loggerQueue) {/*inner sync*/}
```

Reply:
To be fair, if this should ever happen, half of CocoaLumberjack will deadlock 😉

Reply:
@ffried Could you leave a comment about it in this double-sync block? :)

Reply:
I can, but do we need it? There are like hundreds of places in CocoaLumberjack where we first sync the DDLog queue and then the logger's queue.
e.g. `- [DDAbstractLogger logFormatter]`

Reply:
`- [DDFileLogger flush]`
`- [DDFileLogger rollingFrequency]`
`- [DDFileLogger maximumFileSize]`
...

Reply:
Oh well, I've added asserts... ¯\_(ツ)_/¯